packages/dds/tree/src/test/shared-tree/summaryBenchmarkSizes.ts

```diff
@@ -0,0 +1,173 @@
+        assert(summarySize < 20000);
+    });
+    it("with 10 inserted nodes depth-wise.", async () => {
+        const summarySize = await getInsertsSummarySize(10, 0, true);
+        assert(summarySize < 3000);
+    });
+    it("with 100 inserted nodes depth-wise.", async () => {
+        const summarySize = await getInsertsSummarySize(100, 0, true);
+        assert(summarySize < 20000);
+    });
+    it("rejected for 1000 inserts depth wise", async () => {
+        await assert.rejects(getInsertsSummarySize(1000, 0, true), { message: "BatchTooLarge" });
+    });
+});
+
+/**
+ * Inserts a single node under the root of the tree with the given value.
+ */
+function setTestValue(tree: ISharedTree, value: TreeValue, index: number): void {
+    // Apply an edit to the tree which inserts a node with a value
+    tree.runTransaction((forest, editor) => {
+        const writeCursor = singleTextCursor({ type: brand("TestValue"), value });
+        const field = editor.sequenceField(undefined, rootFieldKeySymbol);
+        field.insert(index, writeCursor);
+
+        return TransactionResult.Apply;
+    });
+}
+
+function setTestValueOnPath(tree: ISharedTree, value: TreeValue, path: PlacePath): void {
+    // Apply an edit to the tree which inserts a node with a value.
+    tree.runTransaction((forest, editor) => {
+        const writeCursor = singleTextCursor({ type: brand("TestValue"), value });
+        const field = editor.sequenceField(path, rootFieldKeySymbol);
+        field.insert(0, writeCursor);
+
+        return TransactionResult.Apply;
+    });
+}
+
+function setTestValuesWide(tree: ISharedTree, seed: number, numNodes: number): void {
+    const random = makeRandom(seed);
+    for (let j = 0; j < numNodes; j++) {
+        setTestValue(tree, random.integer(Number.MIN_SAFE_INTEGER, Number.MAX_SAFE_INTEGER), j);
+    }
+}
+
+/**
+ *
+ * @param numNodes - number of nodes you would like to insert
+ * @param seed - seed to randomly generate values for the inserts
+ * @param depth - boolean to specify inserting nodes depth wise or width wise
+ * @returns the byte size of the tree's summary
+ */
+export async function getInsertsSummarySize(
+    numNodes: number,
+    seed: number,
+    depth?: boolean,
+): Promise<number> {
+    const provider = await TestTreeProvider.create(1, true);
+    const tree = provider.trees[0];
+    initializeTestTreeWithValue(tree, 1);
+
+    if (depth) {
+        const fooKey = brand<FieldKey>("foo");
+        const keySet = new Set([fooKey]);
+        setTestValuesNarrow(keySet, seed, tree, numNodes);
+    } else {
+        setTestValuesWide(tree, seed, numNodes);
+    }
+    const summaryTree = await provider.summarize();
+    const summaryString = JSON.stringify(summaryTree);
+    const summarySize = IsoBuffer.from(summaryString).byteLength;
+    return summarySize;
+}
+
+function setTestValuesNarrow(
+    parentKeys: Set<FieldKey>,
+    seed: number,
```

Comment:
I don't think we need to let the caller control the seed since it does not really matter as long as its consistent.

Reply:
hardcoded seed to be 0